modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogManagerImpl.java

```diff
@@ -330,11 +333,30 @@ public CompletableFuture<Void> execute(List<CatalogCommand> commands) {
         return saveUpdateAndWaitForActivation(new BulkUpdateProducer(List.copyOf(commands)));
     }
 
+    /**
+     * Cleanup outdated catalog versions, which can't be observed after given timestamp (inclusively), and compact underlying update log.
+     *
+     * @param timestamp Earliest observable timestamp.
+     * @return Operation future.
+     */
+    public CompletableFuture<Void> compactCatalog(long timestamp) {
+        Catalog catalog = catalogAt(timestamp);
+
+        truncateUpTo(catalog.version(), catalog.time());
```

Comment:
javadoc of `UpdateLog#saveSnapshot` states that snapshotting might not be supported. In that case we will truncate history on only current node, while others will be left intact. Does it make sense to clean the history in only one place -- OnUpdateHandler? I believe this will allow us to preserve consistent behaviour across all the nodes of the cluster